scripts/pylib/twister/twisterlib/size_calc.py

```diff
@@ -160,18 +147,71 @@ def get_rom_size(self):
         """
         return self.rom_size
 
+    def get_percentage_usage_ram(self):
+        """Get the percentage usage of the data that this application uses on device
+
+        @return percentage of RAM.
+        """
+        return self.percentage_usage_ram
+
+    def get_percentage_usage_rom(self):
+        """Get the percentage usage of the data that this application uses on device's flash
+
+        @return percentage of ROM.
+        """
+        return self.percentage_usage_rom
+
     def unrecognized_sections(self):
         """Get a list of sections inside the binary that weren't recognized
 
         @return list of unrecognized section names
         """
         slist = []
-        for v in self.sections:
-            if not v["recognized"]:
-                slist.append(v["name"])
+        if self.filename.endswith(".elf"):
+            for v in self.sections:
+                if not v["recognized"]:
+                    slist.append(v["name"])
         return slist
 
     def _calculate_sizes(self):
```

Comment:
When Twister is used with `--from-buildlog` argument, then the verification of unrecognized sections occurrence is completely disabled and following fail will never occur: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib/reports.py#L417-L421. It should be at least mentioned somewhere in `--from-buildlog` argument help description.

I wondering if the best solution for this problem is execute this elf file scanning every time (and thanks to this sections will be still monitored). And when user would like to use `--from-buildlog`, then `rom` and `ram` sizes would be calculated once again based this time on `build.log`. Thanks to this approach `sections` list will not be empty and could be used to properly verification of `unrecognized_sections`.

Reply:
Done